src/components/productdisplayitem.main.jsx

```diff
@@ -163,6 +168,15 @@ class ProductDisplayItemMain extends React.Component {
     this.setState({ itemQuantity: parseInt(event.target.value, 10) });
   }
 
+  handleConfiguration(configuration, event) {
+    let { itemConfiguration } = this.state;
+    if (!itemConfiguration) {
```

Comment:
Is there ever a point where `itemConfiguration ` would need this check? Looks like your initial value for it in the state is an empty set, which you are fetching.

Reply:
fixed, indeed it cannot happen, I did that on the demo web app, but here I did it properly lol